core/src/main/scala/tech/beshu/ror/accesscontrol/factory/decoders/definitions/ExternalAuthorizationServicesDecoder.scala

```diff
@@ -50,27 +47,89 @@ object ExternalAuthorizationServicesDecoder extends Logging {
+      .decoder
+  }
+
+  private def groupsConfigDecoder(name: ExternalAuthorizationService.Name): Decoder[GroupsConfig] = {
+    SyncDecoderCreator
+      .instance { c =>
+        for {
+          groupIdsConfig <- c.downField("response_group_ids_json_path").as[Option[JsonPath]]
+            .map(_.map(GroupsConfig.GroupIdsConfig.apply))
+          groupIdsDeprecatedConfig <- c.downField("response_groups_json_path").as[Option[JsonPath]]
+            .map(_.map(GroupsConfig.GroupIdsConfig.apply))
+          groupNamesConfig <-
+            c.downField("response_group_names_json_path").as[Option[JsonPath]]
+              .map(_.map(GroupsConfig.GroupNamesConfig.apply))
+        } yield (groupIdsConfig, groupIdsDeprecatedConfig, groupNamesConfig)
+      }
+      .emapE {
+        case (Some(groupIdsConfig), None, groupNamesConfig) => Right(GroupsConfig(groupIdsConfig, groupNamesConfig))
+        case (None, Some(groupIdsDeprecatedConfig), _) => Right(GroupsConfig(groupIdsDeprecatedConfig, None))
+        case (None, None, _) => Left(DefinitionsLevelCreationError(Reason.Message(s"External authorization service '${name.show}' configuration is missing the 'response_group_ids_json_path' attribute")))
+        case (Some(_), Some(_), _) => Left(DefinitionsLevelCreationError(Reason.Message(s"External authorization service '${name.show}' configuration cannot have the 'response_groups_json_path' and 'response_group_ids_json_path' attributes defined at the same time")))
+      }
+      .mapError(DefinitionsLevelCreationError.apply)
+      .decoder
+  }
+
   private implicit val authTokenNameDecoder: Decoder[AuthTokenName] =
     DecoderHelpers.decodeStringLikeNonEmpty.map(AuthTokenName.apply)
 
   private implicit val sendMethodDecoder: Decoder[AuthTokenSendMethod] =
     SyncDecoderCreator
       .from(Decoder.decodeString)
       .emapE[AuthTokenSendMethod] {
-      case "HEADER" => Right(AuthTokenSendMethod.UsingHeader)
-      case "QUERY_PARAM" => Right(AuthTokenSendMethod.UsingQueryParam)
-      case unknown => Left(DefinitionsLevelCreationError(Message(s"Unknown value '$unknown' of 'auth_token_passed_as' attribute")))
-    }
+        case "HEADER" => Right(AuthTokenSendMethod.UsingHeader)
+        case "QUERY_PARAM" => Right(AuthTokenSendMethod.UsingQueryParam)
+        case unknown => Left(DefinitionsLevelCreationError(Message(s"Unknown value '$unknown' of 'auth_token_passed_as' attribute")))
+      }
       .decoder
 
   private implicit val supportedHttpMethodDecoder: Decoder[SupportedHttpMethod] =
     SyncDecoderCreator
       .from(Decoder.decodeString)
       .emapE[SupportedHttpMethod] {
-      case "POST" | "post" => Right(SupportedHttpMethod.Post)
-      case "GET" | "resolve" => Right(Get)
-      case unknown => Left(DefinitionsLevelCreationError(Message(s"Unknown value '$unknown' of 'http_method' attribute")))
-    }
+        case "POST" | "post" => Right(SupportedHttpMethod.Post)
+        case "GET" | "resolve" => Right(Get)
```

Comment:
oh, that's strange. Could you please check (in history) if the "resolve" didn't get here by accident?

Reply:
It was changed from "get" to "resolve" during refactoring. It seems that it was not done on purpose. I restored the previous version.